src/main/java/ch/njol/skript/structures/StructEvent.java

```diff
@@ -0,0 +1,117 @@
+/**
+ *   This file is part of Skript.
+ *
+ *  Skript is free software: you can redistribute it and/or modify
+ *  it under the terms of the GNU General Public License as published by
+ *  the Free Software Foundation, either version 3 of the License, or
+ *  (at your option) any later version.
+ *
+ *  Skript is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU General Public License for more details.
+ *
+ *  You should have received a copy of the GNU General Public License
+ *  along with Skript.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ * Copyright Peter Güttinger, SkriptLang team and contributors
+ */
+package ch.njol.skript.structures;
+
+import ch.njol.skript.Skript;
+import ch.njol.skript.doc.NoDoc;
+import ch.njol.skript.lang.Literal;
+import ch.njol.skript.lang.SkriptEvent;
+import ch.njol.skript.lang.SkriptParser.ParseResult;
+import ch.njol.skript.lang.parser.ParserInstance;
+import org.bukkit.event.Event;
+import org.bukkit.event.EventPriority;
+import org.eclipse.jdt.annotation.Nullable;
+import org.skriptlang.skript.lang.entry.EntryContainer;
+import org.skriptlang.skript.lang.structure.Structure;
+
+import java.util.Locale;
+
+@NoDoc
+public class StructEvent extends Structure {
+
+	static {
+		Skript.registerStructure(StructEvent.class,
+			"[on] <.+> [with priority (:(lowest|low|normal|high|highest|monitor))]");
```

Comment:
Why parse tags over parse marks? Wouldn't it be easier to simply use the mark as an index to access the EventPriority.values() array rather than formatting the tag string and using valueOf? The only downside I can think of is that maybe it'd be a little less robust if the EventPriority enum changed (very unlikely). 

I don't think it's an issue, just wondering if there was a reasoning behind choosing tags.

Reply:
Skript plans to deprecate parse marks in future. I believe that’s the reason behind.

Reply:
I think it was mentioned, but I don't believe it was approved of by the team fully and reasons like what sovde said are partially why

Reply:
> Skript plans to deprecate parse marks in future. I believe that’s the reason behind.

I thought that was just deprecating the broken pipe character. Parse marks are still very handy and less of a hassle than tags for quite a few things, especially when combinations of things are involved.

Reply:
`parser.mark` will remain working but the broken pipe will be deprecated and removed soon. you can use `1:test` and parser.mark will be `1` if matched

Reply:
My bad for mixing up the two.

Reply:
> Why parse tags over parse marks? Wouldn't it be easier to simply use the mark as an index to access the EventPriority.values() array rather than formatting the tag string and using valueOf? The only downside I can think of is that maybe it'd be a little less robust if the EventPriority enum changed (very unlikely).
> 
> I don't think it's an issue, just wondering if there was a reasoning behind choosing tags.

There isn't a specific reason why I chose parse tags, just kind of what I went with 🤷🏻‍♂️ I will change it to parse marks as I think you made a fair point

Reply:
Lol fair enough, I think either method would work fine :)